src/main/scala/cromwell/webservice/CromwellApiHandler.scala

```diff
@@ -0,0 +1,53 @@
+package cromwell.webservice
+
+import akka.actor.{Actor, Props}
+import akka.pattern.ask
+import akka.util.Timeout
+import cromwell.binding._
+import cromwell.engine._
+import cromwell.webservice.CromwellApiHandler.WorkflowStatus
+import cromwell.webservice.PerRequest.RequestComplete
+import spray.http.StatusCodes
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+object CromwellApiHandler {
+  sealed trait WorkflowManagerMessage
+  case class SubmitWorkflow(wdl: WdlSource, inputs: WorkflowInputs) extends WorkflowManagerMessage
+  case class WorkflowStatus(id: WorkflowId) extends WorkflowManagerMessage
+
+  def props(workflowManagerActorProps : Props): Props = {
+    Props(new CromwellApiHandler(workflowManagerActorProps))
+  }
+}
+
+class CromwellApiHandler(workflowManagerActorProps : Props) extends Actor {
+  import context.dispatcher
+
+  override def receive = {
+//    case SubmitWorkflow(wdl, inputs) =>
+//      println(s"Inside the ApiHandler with wdl $wdl")
+//      context.parent ! RequestComplete(StatusCodes.Created, "whatever take 2")
+
+    case WorkflowStatus(id) =>
+      implicit val timeout = Timeout(2.seconds)
+      val handler = context.actorOf(workflowManagerActorProps)
+
+      val future: Future[Option[WorkflowState]] = ask(handler, WorkflowManagerActor.WorkflowStatus(id)).mapTo[Option[WorkflowState]]
+
+      future.onComplete {
+        case Success(state:Option[WorkflowState]) =>
+          if (state.isEmpty) {
```

Comment:
The two most idiomatic ways of doing stuff w/ an Option based on the state are either:

    state match {
      case Some(x) => doSomethingWith(x)
      case None => doSomethingElse
    }

or

    state map {x => doSomethingWith(x)} getOrElse (somethingElse)

The former has the odd distinction of both being viewed as an indicator of n00bness but also championed by Odersky who is certainly not a n00b. (The former is also more performant)